pkgs/net-lib/net/server.rkt

```diff
@@ -0,0 +1,106 @@
+#lang racket/base
+
+(require racket/contract
+         racket/tcp)
+
+(provide
+ (contract-out
+  [start-server (server-proc/c (-> void?))]
+  [run-server (server-proc/c void?)]))
+
+(define (server-proc/c res/c)
+  (->* ((or/c #f string?)
+        listen-port-number?
+        (-> input-port? output-port? any))
+       (#:reuse? any/c
+        #:max-allow-wait exact-nonnegative-integer?
+        #:max-concurrent (or/c +inf.0 exact-positive-integer?)
+        #:listen-proc (-> listen-port-number? exact-nonnegative-integer? any/c (or/c #f string?) evt?)
+        #:accept-proc (-> any/c (values input-port? output-port?))
+        #:close-proc (-> any/c void?)
+        #:timeout-evt-proc (-> thread? input-port? output-port? boolean? evt?))
+       res/c))
+
+(define (start-server host port handle
+                      #:reuse? [reuse? #t]
+                      #:max-allow-wait [max-allow-wait 4]
+                      #:max-concurrent [max-concurrent +inf.0]
+                      #:listen-proc [listen tcp-listen]
+                      #:accept-proc [accept tcp-accept]
```

Comment:
Do we need an argument like `tcp-accept/enable-break`? I just saw that the old `run-server` had both: I haven't refreshed my memory as to why yet.

Reply:
Not with this implementation, since the job of allowing breaks within the listening thread is handled by the `sync/enable-break` call instead.